mrp_mto_analytic/models/stock_rule.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2024 Quartile Limited
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class StockRule(models.Model):
+    _inherit = "stock.rule"
+
+    def _prepare_mo_vals(
+        self,
+        product_id,
+        product_qty,
+        product_uom,
+        location_dest_id,
+        name,
+        origin,
+        company_id,
+        values,
+        bom,
+    ):
+        vals = super(StockRule, self)._prepare_mo_vals(
+            product_id,
+            product_qty,
+            product_uom,
+            location_dest_id,
+            name,
+            origin,
+            company_id,
+            values,
+            bom,
+        )
+        move_dest_ids = values.get("move_dest_ids")
+        if move_dest_ids:
+            vals["analytic_distribution"] = move_dest_ids[
+                0
+            ].raw_material_production_id.analytic_distribution
```

Comment:
```suggestion
            vals["analytic_distribution"] = move_dest_ids[0].analytic_distribution
```
Wouldn't it technically make more sense to do this, and switch the dependency to `mrp` and `stock_analytic`, removing `mrp_stock_analytic`?

Reply:
The feature of this module will not work without `mrp_stock_analytic`. I mean `stock.move` that comes from MO will not have values for `analytic_distribution`. IMO, we should not remove `mrp_stock_analytic` from the dependency.

Reply:
My previous comment was invalid as we need `mrp_stock_analytic` to have `analytic_distribution` in `mrp.production`.